src/main/java/de/rwth/idsg/steve/web/validation/ChargeBoxIdValidator.java

```diff
@@ -40,4 +43,22 @@ public void initialize(ChargeBoxId idTag) {
     public boolean isValid(String string, ConstraintValidatorContext constraintValidatorContext) {
         return string == null || PATTERN.matcher(string).matches();
     }
+
+    public boolean isValid(String str) {
+        if (Strings.isNullOrEmpty(str)) {
+            return false;
+        }
+
+        String str1 = str.strip();
+        if (!str1.equals(str)) {
```

Comment:
your new version is the demonstration of what i was trying to prevent: an unreadable complexity hard to understand.

dont get me wrong. regex is a powerful machine, but too much of it can cause head aches. therefore, i try to minimize the use of it and KISS.

Reply:
maybe you wanted to give this decision (whether spaces at at start and end are allowed or not) to the user, by making this a part of the regex which can be overridden?

if yes: i think i am making a reasonable opinionated decision here by not allowing them.

if no: then, we are just talking about refactor. i argue that simple java code is simpler than complex regex.

Reply:
My point is still the same: if it is not configurable, it will be dangerous for some users and you may lost some chargers without any simple way to restore them.

I agree that regex may be sometimes difficult to read but you rarely do it and tests are here to validate the expected behavior.

I think the tradeoff is acceptable here.

Otherwise, another configuration is possible too.

Reply:
I missed that spaces were not allowed before then forget my point.